File: fc-chat/demo/part3/ch3/02-4-message-system-add-connection/src/main/java/net/prostars/messagesystem/handler/websocket/BaseRequestHandler.java
package net.prostars.messagesystem.handler.websocket;

import net.prostars.messagesystem.dto.websocket.inbound.BaseRequest;
import org.springframework.web.socket.WebSocketSession;

public interface BaseRequestHandler<T extends BaseRequest> {
  void handleRequest(WebSocketSession webSocketSession, T request);
}

File: fc-chat/demo/part3/with-push/src/main/kotlin/com/yscorp/withpush/messagesystem/handler/websocket/BaseRequestHandler.kt
package com.yscorp.withpush.messagesystem.handler.websocket

import com.yscorp.withpush.messagesystem.dto.websocket.inbound.BaseRequest
import org.springframework.web.socket.WebSocketSession

interface BaseRequestHandler<T : BaseRequest> {
    fun handleRequest(webSocketSession: WebSocketSession, request: T)
}
